base/strings/search.jl

```diff
@@ -125,28 +123,14 @@ function _searchindex(s::Union{AbstractString,ByteArray},
         return 1 <= i <= nextind(s,lastindex(s)) ? i :
                throw(BoundsError(s, i))
     end
-    t1, j2 = next(t,start(t))
+    t1, trest = Iterators.peel(t)
     while true
         i = findnext(equalto(t1),s,i)
         if i === nothing return 0 end
-        c, ii = next(s,i)
-        j = j2; k = ii
-        matched = true
-        while !done(t,j)
-            if done(s,k)
-                matched = false
-                break
-            end
-            c, k = next(s,k)
-            d, j = next(t,j)
-            if c != d
-                matched = false
-                break
-            end
-        end
-        if matched
-            return i
-        end
+        ii = nextind(s, i)
+        a = Iterators.Stateful(trest)
+        matched = all(splat(==), zip(s[ii:end], a))
```

Comment:
`SubString(z, ii)`

Reply:
Same thing, indexing into a string gives a SubString

Reply:
Never mind, I'm wrong about this.